File: gdk/Models/AccountType.swift
public enum AccountType: String, CaseIterable, Codable, Comparable, Equatable {
    /// multiSig
    case standard = "2of2"
    case amp = "2of2_no_recovery"
    case twoOfThree = "2of3"

    /// singleSig
    case legacy = "p2pkh"
    case segwitWrapped = "p2sh-p2wpkh" // former legacy
    case segWit = "p2wpkh"
    case taproot = "p2tr"
    case lightning = "lightning"
    
    public var multisig: Bool {
        [AccountType.standard, AccountType.amp, AccountType.twoOfThree].contains(self)
    }

    public var singlesig: Bool {
        [.legacy, AccountType.segwitWrapped, AccountType.segWit, AccountType.taproot].contains(self)
    }

    public var lightning: Bool {
        AccountType.lightning == self
    }

    public var string: String {
        get {
            switch self {
            case .standard:
                return "2FA Protected"
            case .amp:
                return "AMP"
            case .twoOfThree:
                return "2of3 with 2FA"
            case .legacy:
                return "Legacy"
            case .segwitWrapped:
                return "Legacy SegWit"
            case .segWit:
                return "Standard"
            case .taproot:
                return "Taproot"
            case .lightning:
                return "Lightning"
            }
        }
    }

    public var shortString: String {
        get {
            switch self {
            case .standard:
                return "2of2"
            case .amp:
                return "Amp"
            case .twoOfThree:
                return "2of3"
            case .legacy:
                return "Legacy"
            case .segwitWrapped:
                return "Legacy SegWit"
            case .segWit:
                return "Native SegWit"
            case .taproot:
                return "Taproot"
            case .lightning:
                return "Lightning"
            }
        }
    }

    public static func < (a: AccountType, b: AccountType) -> Bool {
        let rules: [AccountType] = [.segwitWrapped, .segWit, .taproot, .standard, .amp, .twoOfThree, .lightning]
        return rules.firstIndex(of: a) ?? 0 < rules.firstIndex(of: b) ?? 0
    }
}

public enum RecoveryKeyType {
    case hw
    case newPhrase
    case existingPhrase
    case publicKey
}

File: data/src/commonMain/kotlin/com/blockstream/data/gdk/data/AccountType.kt
package com.blockstream.data.gdk.data

import kotlinx.serialization.Serializable

@Serializable
enum class AccountType(val gdkType: String) {
    // Mutlisig
    STANDARD("2of2"),
    AMP_ACCOUNT("2of2_no_recovery"),
    TWO_OF_THREE("2of3"),

    // Singlesig
    BIP44_LEGACY("p2pkh"),
    BIP49_SEGWIT_WRAPPED("p2sh-p2wpkh"),
    BIP84_SEGWIT("p2wpkh"),
    BIP86_TAPROOT("p2tr"),

    LIGHTNING("lightning"),

    UNKNOWN("unknown");

    override fun toString(): String = when (this) {
        BIP44_LEGACY -> "Legacy"
        BIP49_SEGWIT_WRAPPED -> "Legacy SegWit"
        BIP84_SEGWIT -> "Standard"
        BIP86_TAPROOT -> "Taproot"
        LIGHTNING -> "Lightning"

        STANDARD -> "2FA Protected"
        AMP_ACCOUNT -> "AMP"
        TWO_OF_THREE -> "2of3 with 2FA"
        else -> gdkType
    }

    fun isSinglesig() = when (this) {
        BIP44_LEGACY, BIP49_SEGWIT_WRAPPED, BIP84_SEGWIT, BIP86_TAPROOT -> true
        else -> false
    }

    fun isLightning() = this == LIGHTNING

    fun isMutlisig() = !isSinglesig() && !isLightning()

    companion object {
        fun byGDKType(name: String) = when (name) {
            "2of2" -> STANDARD
            "2of2_no_recovery" -> AMP_ACCOUNT
            "2of3" -> TWO_OF_THREE
            "p2pkh" -> BIP44_LEGACY
            "p2sh-p2wpkh" -> BIP49_SEGWIT_WRAPPED
            "p2wpkh" -> BIP84_SEGWIT
            "p2tr" -> BIP86_TAPROOT
            "lightning" -> LIGHTNING
            else -> UNKNOWN
        }
    }
}